src/cmem/ActivityControl/SilkActivityControl.tsx

```diff
@@ -167,38 +186,125 @@ export function useSilkActivityControl({
     layoutConfig = defaultLayout,
     hideMessageOnStatus = () => false,
     executePrioritized,
+    translateUnits, 
     ...props
 }: SilkActivityControlProps) {
     const [activityStatus, setActivityStatus] = useState<SilkActivityStatusProps | undefined>(initialStatus);
-    const currentStatus = useRef<SilkActivityStatusProps | undefined>(initialStatus)
-    const [showStartPrioritized, setShowStartPrioritized] = useState(false)
+    const currentStatus = useRef<SilkActivityStatusProps | undefined>(initialStatus);
+    const [showStartPrioritized, setShowStartPrioritized] = useState(false);
     const [errorReport, setErrorReport] = useState<string | SilkActivityExecutionReportProps | undefined>(undefined);
+    const [statusTimers, setStatusTimers] = React.useState<StatusTimers>({ ...defaultStatusTimers });
+    const runningTimerRef = React.useRef<NodeJS.Timer>();
+    const waitingTimerRef = React.useRef<NodeJS.Timer>();
 
-    // Register update function
-    useEffect(() => {
-        const updateActivityStatus = (status: SilkActivityStatusProps | undefined) => {
-            if(status?.concreteStatus !== "Waiting") {
-                setShowStartPrioritized(false)
-            } else if(executePrioritized) {
-                // Show start prioritized button only-if the activity is still in Waiting status after 2s
-                setTimeout(() => {
-                    if(currentStatus.current?.concreteStatus === "Waiting") {
-                        setShowStartPrioritized(true)
-                    }
-                }, 2000)
+
+    /** start counting time from now or from when it was started in the case of a reload */
+    const tickForStatusTimers = React.useCallback(
+        (timer: "runningTime" | "waitingTime", status?: SilkActivityStatusProps) => {
+            if (!status || !translateUnits) return;
+            const { startTime, queueTime } = status;
+            const isRunningTimer = timer === "runningTime";
+            const timerRef = isRunningTimer ? runningTimerRef : waitingTimerRef;
+            if (!timerRef.current) {
+                //then it's most likely that it's a reload or fresh start.
+                const initialDiff =
```

Comment:
This can be replaced by `ElapsedDateTimeDisplay`, either directly in `ActivityControlWidget` by only giving it the start time of the timer, or by giving it the `ElapsedDateTimeDisplay` element directly.